src/logger/auto_logger.c

```diff
@@ -50,147 +45,74 @@
-        if (!tok)
-                return;
-
-        jsmn_exists_set_val_float(tok, "speed", &alst->speed);
-        jsmn_exists_set_val_int(tok, "time", &alst->time);
-}
-
-
 bool auto_logger_set_config(struct auto_logger_config* cfg,
                             const jsmntok_t *json)
 {
-        jsmn_exists_set_val_bool(json, "active", &cfg->active);
-        set_speed_time(&cfg->start, "start", json);
-        set_speed_time(&cfg->stop, "stop", json);
-        return true;
-}
-
-bool auto_logger_init(struct auto_logger_config* cfg)
-{
-        if (!cfg)
-                return false;
-
-        auto_logger_state.cfg = cfg;
-        auto_logger_state.logging = logging_is_active();
-        auto_logger_state.timestamp_start = 0;
-        auto_logger_state.timestamp_stop = 0;
+        jsmn_exists_set_val_bool(json, "en", &cfg->enabled);
+        jsmn_exists_set_val_string(json, "channel", cfg->channel, DEFAULT_LABEL_LENGTH, true);
+        set_auto_control_trigger(&cfg->start, "start", json);
+        set_auto_control_trigger(&cfg->stop, "stop", json);
         return true;
 }
 
-static bool should_start_logging(const GpsSample* sample,
-                                 const tiny_millis_t uptime)
+static void auto_logger_sample_cb(const struct sample* sample,
+                           const int tick, void* data)
 {
-        const tiny_millis_t trig_time =
-                (tiny_millis_t) auto_logger_state.cfg->start.time * 1000;
-        const float trig_speed = auto_logger_state.cfg->start.speed;
-
-        if (0 == trig_time)
-                return false;
+        if (!auto_logger_state.cfg || !auto_logger_state.cfg->enabled)
+                return;
 
-        if (sample->speed < trig_speed) {
-                auto_logger_state.timestamp_start = 0;
-                return false;
-        }
+        double value;
+        if (!get_sample_value_by_name(sample, auto_logger_state.cfg->channel, &value))
+                return;
 
-        if (0 == auto_logger_state.timestamp_start) {
-                auto_logger_state.timestamp_start = uptime;
-                return false;
+        enum auto_control_trigger_result res = auto_control_check_trigger(value,
+                                                                          &auto_logger_state.cfg->start,
+                                                                          &auto_logger_state.cfg->stop,
+                                                                          &auto_logger_state.control_state);
+        switch(res){
+                case AUTO_CONTROL_TRIGGERED:
+                        startLogging();
+                        pr_info(LOG_PFX "Auto-starting logging\r\n");
+                        break;
+                case AUTO_CONTROL_UNTRIGGERED:
+                        stopLogging();
+                        pr_info(LOG_PFX "Auto-stopping logging\r\n");
+                        break;
+                default:
```

Comment:
Unnecessary because if there is no match it will just skip the switch statement.

Reply:
Right. I think I did it to make the compiler happy, but maybe it was to make the IDE warnings happy. I'll remove it if it does not fail the build.  Thanks.